examples/ticks_and_spines/tick-formatters.py

```diff
@@ -3,52 +3,51 @@
 Tick formatters
 ===============
 
-Show the different tick formatters.
+Tick formatters define how the numeric value associated with a tick on an axis
+is formatted as a string.
+
+This example illustrates the usage and effect of the most common formatters.
 """
 
 import numpy as np
 import matplotlib.pyplot as plt
 import matplotlib.ticker as ticker
 
 
-# Setup a plot such that only the bottom spine is shown
-def setup(ax):
+def setup(ax, title):
+    """Set up common parameters for the Axes in the example."""
+    # only show the bottom spine
+    ax.yaxis.set_major_locator(ticker.NullLocator())
     ax.spines['right'].set_color('none')
     ax.spines['left'].set_color('none')
-    ax.yaxis.set_major_locator(ticker.NullLocator())
     ax.spines['top'].set_color('none')
+
+    # define tick positions
+    ax.xaxis.set_major_locator(ticker.MultipleLocator(1.00))
+    ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
+
     ax.xaxis.set_ticks_position('bottom')
     ax.tick_params(which='major', width=1.00, length=5)
     ax.tick_params(which='minor', width=0.75, length=2.5, labelsize=10)
     ax.set_xlim(0, 5)
     ax.set_ylim(0, 1)
-    ax.patch.set_alpha(0.0)
+    ax.text(0.0, 0.2, title, transform=ax.transAxes,
+            fontsize=14, fontname='Monospace', color='tab:blue')
 
 
-fig = plt.figure(figsize=(8, 6))
-n = 7
+fig, axs = plt.subplots(7, 1, figsize=(8, 6))
 
 # Null formatter
-ax = fig.add_subplot(n, 1, 1)
-setup(ax)
-ax.xaxis.set_major_locator(ticker.MultipleLocator(1.00))
-ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
-ax.xaxis.set_major_formatter(ticker.NullFormatter())
-ax.xaxis.set_minor_formatter(ticker.NullFormatter())
-ax.text(0.0, 0.1, "NullFormatter()", fontsize=16, transform=ax.transAxes)
+setup(axs[0], title="NullFormatter()")
+axs[0].xaxis.set_major_formatter(ticker.NullFormatter())
 
 # Fixed formatter
-ax = fig.add_subplot(n, 1, 2)
-setup(ax)
-ax.xaxis.set_major_locator(ticker.MultipleLocator(1.0))
-ax.xaxis.set_minor_locator(ticker.MultipleLocator(0.25))
```

Comment:
One of the most common mistakes people make is to use a `FixedFormatter` (e.g. through `set_xticklabels`) without defining the ticks' locations). For this specific example it is very misleading to hide the locator inside some `setup`, because it's not clear on first sight why those ticklabels are correct.

Reply:
I've now explicitly set the locator before the fixed formatter and added a comment why one should do so.